packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2853,10 +2864,6 @@ class _MenuPanelState extends State<_MenuPanel> {
     }
     return ConstrainedBox(
       constraints: effectiveConstraints,
-      child: UnconstrainedBox(
-        constrainedAxis: widget.orientation,
-        clipBehavior: Clip.hardEdge,
-        alignment: AlignmentDirectional.centerStart,
```

Comment:
Why is the `UnconstrainedBox` no longer needed here?

Reply:
If the menu direction is vertical, like the menu of the combo box, the `UnconstrainedBox` ignores the constraints on the cross-axis, and we cannot use the `menuStyle` to control the size of the menu. That's why I was thinking maybe we should just remove it. But it is a little tricky to handle the clipping behavior in other places (tried some widgets `ClipRect` or `Wrap` but cannot make it work in an ideal way.). So instead of removing this part, I just added one more property `crossAxisUnconstrained` in `MenuAnchor` and make it default to true. Only when we need to constrain the menu in both axes, we can turn this flag to false.

This probably is not the best way to handle the issue, but at least it will not change the old behaviors/tests and it works on the combo box. But please do let me know if there's a better way! Thanks a lot!